src/set_partner_key.c

```diff
@@ -73,11 +73,14 @@ int set_partner_key(const command_t *cmd) {
         return reply_error(err);
     }
 
-    // The incoming partner name is NOT NULL terminated, so we use memcpy and
-    // manually NULL terminate the buffer
-    memset(G_swap_ctx.partner.name, 0, sizeof(G_swap_ctx.partner.name));
-    memcpy(G_swap_ctx.partner.name, partner.bytes, partner.size);
-    G_swap_ctx.partner.name_length = partner.size;
+    memset(G_swap_ctx.partner.prefixed_name, 0, sizeof(G_swap_ctx.partner.prefixed_name));
+    // Prepare the prefix for FUND display
+    strlcpy(G_swap_ctx.partner.prefixed_name,
+            PARTNER_NAME_PREFIX_FOR_FUND,
+            sizeof(G_swap_ctx.partner.prefixed_name));
+    // The incoming partner name is NOT NULL terminated, so we use strncat
+    // Don't erase the prefix copied above
+    strncat(G_swap_ctx.partner.prefixed_name, (char *) partner.bytes, partner.size);
```

Comment:
Do we have a check somewhere that `partner.size <= MAX_PARTNER_NAME_LENGHT`?

Reply:
Yes, the parser also checks boundaries